src/tests/Equinor.ProCoSys.IPO.Command.Tests/InvitationCommands/UpdateRfocVoided/UpdateRfocVoidedCommandHandlerTests.cs

```diff
@@ -0,0 +1,186 @@
+
+            _loggerMock = new Mock<ILogger<UpdateRfocVoidedCommandHandler>>();
+
+            //create invitation
+            _invitation = new Invitation(
+                    _plant,
+                    _project,
+                    _title,
+                    _description,
+                    _typeDP,
+                    new DateTime(),
+                    new DateTime(),
+                    null,
+                    new List<McPkg> { new McPkg(_plant, _project, "CommNo1", "McNo1", "d", "1|2"), new McPkg(_plant, _project, "CommNo1", "McNo2", "d", "1|2") },
+                    null);
+            _invitation.ScopeHandedOver();
+
+            //command
+            _command = new UpdateRfocVoidedCommand(_projectName, _certificateGuid);
+
+            _dut = new UpdateRfocVoidedCommandHandler(
+                _invitationRepositoryMock.Object,
+                _projectRepositoryMock.Object,
+                _unitOfWorkMock.Object,
+                _plantProviderMock.Object,
+                _certificateApiServiceMock.Object,
+                _certificateRepositoryMock.Object,
+                _loggerMock.Object);
+        }
+
+        [TestMethod]
+        public async Task HandlingUpdateRfocVoidedCommand_ShouldCallMainApi()
+        {
+            Assert.AreEqual(IpoStatus.ScopeHandedOver, _invitation.Status);
+
+            var result = await _dut.Handle(_command, default);
+
+            Assert.AreEqual(ServiceResult.ResultType.Ok, result.ResultType);
+
+            _certificateApiServiceMock.Verify(c => c.GetCertificateMcPkgsAsync(_plant, _certificateGuid), Times.Once);
+            _certificateApiServiceMock.Verify(c => c.GetCertificateCommPkgsAsync(_plant, _certificateGuid), Times.Once);
+        }
+
+        [TestMethod]
+        public async Task HandlingUpdateRfocVoidedCommand_ShouldCallSaveChanges()
+        {
+            var result = await _dut.Handle(_command, default);
+
+            Assert.AreEqual(ServiceResult.ResultType.Ok, result.ResultType);
+
+            _unitOfWorkMock.Verify(t => t.SaveChangesAsync(It.IsAny<CancellationToken>()), Times.Once);
+        }
+
+        [TestMethod]
+        public async Task HandlingUpdateRfocVoidedCommand_ShouldNotSaveChangesWhenVoidedCertificateIsNotActuallyVoidedOrDeleted()
+        {
+            _certificateApiServiceMock
+                .Setup(c => c.GetCertificateCommPkgsAsync(_plant, _certificateGuid))
+                .Returns(Task.FromResult(_certificateCommPkgsModel));
+            _certificateApiServiceMock
+                .Setup(c => c.GetCertificateMcPkgsAsync(_plant, _certificateGuid))
+                .Returns(Task.FromResult(_certificateMcPkgsModel));
+
+            var result = await _dut.Handle(_command, default);
+
+            Assert.AreEqual(ServiceResult.ResultType.Unexpected, result.ResultType);
+
+            _unitOfWorkMock.Verify(t => t.SaveChangesAsync(It.IsAny<CancellationToken>()), Times.Never);
+        }
+    }
+}
```

Comment:
I should like to see some more positive tests. A test should proof setting cerificate.IsVoided, a second test should proof caalling _invitationRepository.RfocVoidedHandling,

Reply:
Add even more tests: Test early exit when project not found and if project is closed.
You can assert that getting cerificate is never done

Also add unit tests on different UnexpectedResult-exists